src/reverse.js

```diff
@@ -1,24 +1,33 @@
 var _curry1 = require('./internal/_curry1');
+var _isString = require('./internal/_isString');
 var _slice = require('./internal/_slice');
 
 
 /**
- * Returns a new list with the same elements as the original list, just
- * in the reverse order.
+ * Returns a new list or string with the elements or characters in reverse
+ * order.
  *
  * @func
  * @memberOf R
  * @category List
  * @sig [a] -> [a]
- * @param {Array} list The list to reverse.
- * @return {Array} A copy of the list in reverse order.
+ * @sig String -> String
+ * @param {Array|String} list
+ * @return {Array|String}
  * @example
  *
  *      R.reverse([1, 2, 3]);  //=> [3, 2, 1]
  *      R.reverse([1, 2]);     //=> [2, 1]
  *      R.reverse([1]);        //=> [1]
  *      R.reverse([]);         //=> []
+ *
+ *      R.reverse('abc');      //=> 'cba'
+ *      R.reverse('ab');       //=> 'ba'
+ *      R.reverse('a');        //=> 'a'
+ *      R.reverse('');         //=> ''
  */
 module.exports = _curry1(function reverse(list) {
-  return _slice(list).reverse();
+  return _isString(list) ?
+    list.split('').reverse().join('') :
+    _slice(list).reverse();
```

Comment:
I believe that relies on the environment supporting bracket notation for character access. It doesn't work for arrays, though, as one ends up doing `R.concat([], 3)` when reversing `[1, 2, 3]`. This is currently a type error, but interestingly it will work if and when #1272 is merged. Relying on `[] ++ 3` working would feel icky, of course!